### ISSUE TITLE
sst-test-core doesn't detect python name

### ISSUE THREAD

**Issue by calewis** [sstsimulator/sst-core#681]
I compiled sst-core with python3. The build and install seemly go okay, but when I try to run `sst-test-core` I get: 
```
bash-5.1# /install/bin/sst-test-core
env: can't execute 'python': No such file or directory
```

**Comment by allevin** [comment]
@calewis The problem is with the shebang of the launch script file `core-root/src/sst/core/testingframework/sst-test-core`.  

The shbang text is `#!/usr/bin/env python` which should be python2, but on some OS distros will make "python" point at Python3.   The problem is if we change the shebang to `#!/usr/bin/env python3` then we will break the script for python2 support.   This is a very tricky problem that the entire python community has been struggling with.

**Comment by calewis** [comment]
@allevin it appears to me that sst_check_python.m4 determines a python_exe to associate with a build/install of core (the build doesn't fail even though the system only has python3).  If the SST build already detected that it is using `python3` instead of `python`, why can't we just use the python exe found by configure to run the tests?

**Comment by allevin** [comment]
@calewis the test frameworks runs completely independent of how SST is built and runs (in fact you can build SST with Py3  and test it using Py2).   It uses Python unittest to launch SST and then does analysis of the output.  It does not care about the specifics of how SST runs or is built.   

The launch scripts (`sst-test-core` and `sst-test-elements`) are very simplistic python scripts that and intended to load up the main test engine which does all the testing.   They are both set as executable scripts so the user does not have to run `python -m sst-test-core`   At the point where you call `sst-test-core` it is basically like any other shell script that uses the shebang to decide how to interpret the script.  At launch it knows nothing about SST.  

Later, after the test engine is loaded, there are many functions available to the developer to discover how SST is built and configured which could do what you are proposing, but that information is not available when `sst-test-core` is initially called.   

You should be able to launch the test using `python3 <path_to_sst_root_install>/bin/sst-test-core`.  Again, this is a problem with inconsistent Python naming on OS's are along with shell script shebang's. Once we abandon Py2 support, this will be much cleaner across the board.

**Comment by nmhamster** [comment]
We could generate the script during the build, after we have figured out the Python right?

**Comment by allevin** [comment]
@nmhamster I assume we could, but it might be a fairly head scratching effort to figure out.  Right now, the file is static and exists in the core directory and is copied over to the bin directory during a make install.  We would have to mod the build system to change the launch scripts dynamically before the make install is called (during the configure step?).

**Comment by calewis** [comment]
I think it's reasonable that anything that lands in /bin of the install directory should use the python that core was configured with.  If you expect users to invoke it with `python sst-test-core.py` why make it an executable in bin?

**Comment by nmhamster** [comment]
@calewis / @allevin - please can you both take a look at PR #683. This utilizes the `PYTHON_EXE` that is detected to auto-populate `sst-test-core` and `sst-test-element` scripts that are installed in `bin`.

**Comment by calewis** [comment]
Fixed with #683
